pkgs/development/python-modules/redshift-connector/default.nix

```
@@ -0,0 +1,58 @@
+{ beautifulsoup4
+, boto3
+, buildPythonPackage
+, fetchFromGitHub
+, lib
+, lxml
+, packaging
+, pytest
+, pytest-cov
+, pytest-mock
+, pytestCheckHook
+, pythonOlder
+, pytz
+, requests
+, scramp
+}:
+
+buildPythonPackage rec {
+  pname = "redshift-connector";
+  version = "2.0.910";
+
+  disabled = pythonOlder "3.6";
+
+  src = fetchFromGitHub {
+    owner = "aws";
+    repo = "amazon-redshift-python-driver";
+    rev = "v${version}";
+    hash = "sha256-24yI6pXSHxhT30N3rJXAMtpCOhhGsBuDrwx9jMO1FW0=";
+  };
+
+  propagatedBuildInputs = [
+    beautifulsoup4
+    boto3
+    lxml
+    packaging
+    pytz
+    requests
+    scramp
+  ];
+
+  nativeCheckInputs = [ pytestCheckHook ];
+
+  checkInputs = [
+    pytest
+    pytest-cov
+    pytest-mock
+  ];
```

```suggestion
  nativeCheckInputs = [
    pytest-mock
    pytestCheckHook
  ];
```

I think all should be `nativeCheckInputs`. We also don't want `pytest-cov`, which checks for coverage, which just isn't a useful metric for us downstream.

Then remove `--cov` arguments;

```
  postPatch = ''
    sed -i "/--cov/d" setup.cfg
  '';

---

Thanks! There are no `nativeBuildInputs` here, so I was thinking to move this before `propagatedBuildInputs` (where `nativeBuildInputs` would be if it existed). Or do you mean to put it after `nativeCheckInputs` in this case?

---

No, your first assumption is correct.